src/core/did.ts

```diff
@@ -0,0 +1,12 @@
+import * as dns from '../dns'
+
+export const rootDIDForUser = async (
+  username: string,
+  domain = 'fission.name'
+): Promise<string> => {
+  try {
+    return dns.lookupTxtRecord(`_did.${username}.${domain}`)
+  } catch (err) {
+    throw new Error("Could not locate user DID in dns")
+  }
```

Comment:
Nice, was just about to add this as well 👍 
Just an FYI, I just improved `dns.lookupTxtRecord` a bit, now joins multiple TXTs, but now it can return `null` as well.
So instead of try catch, you'll need to do a `null` check.

Nitpick, should we standardise on uppercase DID or camelcase in func/var names? 🤔 I've been doing camelcase like `lookupTxtRecord` instead of `lookupTXTRecord`. Slightly prefer camelcase, but fine with either.

PS. Just realised I haven't been using the `Maybe` type, sorry about that, need to get on that 👀

Reply:
good looks^^ I'll fix that on rebase

> Nitpick, should we standardise on uppercase DID or camelcase in func/var names? thinking I've been doing camelcase like lookupTxtRecord instead of lookupTXTRecord. Slightly prefer camelcase, but fine with either

Yeah I think I like camelcase better actually. I'll start standardizing around that :ok_hand: